pkg/agent/controller/networkpolicy/fqdn_test.go

```diff
@@ -584,3 +590,159 @@ func TestSyncDirtyRules(t *testing.T) {
+			},
+			expectedIPs: map[string]ipWithExpiration{
+				"192.1.1.1": {ip: net.ParseIP("192.1.1.1"), expirationTime: currentTime.Add(5 * time.Second)},
+				"192.1.1.2": {ip: net.ParseIP("192.1.1.2"), expirationTime: currentTime.Add(5 * time.Second)},
+			},
+			expectedRequeryAfter: ptr.To(5 * time.Second),
+		},
+		{
+			name: "stale IP with expired TTL is evicted",
+			existingDNSCache: map[string]dnsMeta{
+				testFQDN: {
+					responseIPs: map[string]ipWithExpiration{
+						"192.1.1.1": {ip: net.ParseIP("192.1.1.1"), expirationTime: currentTime.Add(-1 * time.Second)},
+					},
+				},
+			},
+			dnsResponseIPs: map[string]ipWithExpiration{
+				"192.1.1.3": {ip: net.ParseIP("192.1.1.3"), expirationTime: currentTime.Add(5 * time.Second)},
+			},
+			expectedIPs: map[string]ipWithExpiration{
+				"192.1.1.3": {ip: net.ParseIP("192.1.1.3"), expirationTime: currentTime.Add(5 * time.Second)},
+			},
+			expectedRequeryAfter: ptr.To(5 * time.Second),
+		},
+		{
+			name:             "existingDNSCache is empty, the new response matches a selector.",
+			existingDNSCache: map[string]dnsMeta{},
+			dnsResponseIPs: map[string]ipWithExpiration{
+				"192.1.1.1": {ip: net.ParseIP("192.1.1.1"), expirationTime: currentTime.Add(5 * time.Second)},
+			},
+			expectedIPs: map[string]ipWithExpiration{
+				"192.1.1.1": {ip: net.ParseIP("192.1.1.1"), expirationTime: currentTime.Add(5 * time.Second)},
+			},
+			expectedRequeryAfter: ptr.To(5 * time.Second),
+			mockSelectorToRuleIDs: map[fqdnSelectorItem]sets.Set[string]{
+				selectorItem1: sets.New[string]("mockRule1"),
+			},
+		},
+		{
+			name:             "existingDNSCache is empty, the new response doesn't match any selector",
+			existingDNSCache: map[string]dnsMeta{},
+			dnsResponseIPs: map[string]ipWithExpiration{
+				"192.1.1.1": {ip: net.ParseIP("192.1.1.1"), expirationTime: currentTime.Add(5 * time.Second)},
+			},
+			expectedIPs: nil,
+			mockSelectorToRuleIDs: map[fqdnSelectorItem]sets.Set[string]{
+				selectorItem2: sets.New[string]("mockRule2"),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			fakeClock := newFakeClock(currentTime)
+			controller := gomock.NewController(t)
+			f, _ := newMockFQDNController(t, controller, nil, fakeClock)
+			f.dnsEntryCache = tc.existingDNSCache
+			if tc.mockSelectorToRuleIDs != nil {
+				f.selectorItemToRuleIDs = tc.mockSelectorToRuleIDs
+			}
+			require.Zero(t, fakeClock.TimersAdded())
+
+			f.onDNSResponse(testFQDN, tc.dnsResponseIPs, nil)
+
+			cachedDnsMetaData, _ := f.dnsEntryCache[testFQDN]
+
+			assert.Equal(t, tc.expectedIPs, cachedDnsMetaData.responseIPs, "FQDN cache doesn't match expected entries")
+
+			if tc.expectedRequeryAfter != nil {
+				// Wait for the DelayingQueue to create the timer which signals that the
+				// DNS request for the FQDN item is ready to be sent.
+				require.Eventually(t, func() bool {
+					return fakeClock.TimersAdded() > 0
+				}, 1*time.Second, 10*time.Millisecond)
```

Comment:
nit: could you make it a one-liner to be consistent with the `require.Eventually` below?

Reply:
Done.